zulipterminal/ui_tools/views.py

```diff
@@ -1192,6 +1192,11 @@ def __init__(
             # urwid.Pile(msg_box.footer),
         )
 
+    def _handle_file_upload(self, file_location):
+        file_location1 = file_location
+        self.uri = self.model.get_file_upld_uri(file_location1)
+        self.controller.set_uri(self.uri)  # Update the uri in the controller
```

Comment:
This function can be called when `keypress()` detects `ENTER` and the popup can be closed upon encountering this event.